R/combineSNV_CNA.R

```diff
@@ -0,0 +1,73 @@
+# combineSNV_CNA.R
+
+#' Combine mutation data from CNA and SNV
+#'
+#' \code{combineSNV_CNA} loads the vector of preprocessed dataset from filtered CNA or SNV files,
+#'  creates hash keys corresponding to the input file type, combine and save the hashed
+#'  keys to a gX format file as output file.
+#'
+#' Details.
+#' @section gX files:
+#'  gX data frame files contain data of SNV and CNA combined for each gene as one object, sorted per
+#'  gene.
+#'
+#'
+#' @param fName A vector of path of fully qualified file names of SNV and/or CNA files.
+#' @param fgX The fully qualified filename/path of a gX output file, by default is gX.rds.
+#' @param silent Boolean option for writing combiling process information to console, FALSE by default.
+#' @param writeLog Boolean option for log results, TRUE by default.
+#'
+#'
+#' @examples
+#' \dontrun{combine(fName=file_names, fgX="combined_data.rds", silent=TRUE, writeLog=FALSE)}
+#'
+#' @export
+combineSNV_CNA <- function(fname, fgX = "gX.rds", silent=FALSE, writeLog=TRUE) {
+    #' pseudo code
+    #'
+    #' using hash package, e.g extract keys
+    #' if(!require(hash)){
+    #'  install.packages("hash")
+    #'  library(hash)
+    #' }
+    #' hashTable <- new.env()
+    #' read from input files and store hash
+    #' for (i <- 1, i <= length(fname), i++) {
+    #'  fcurrent <- fname[i]
+    #'  if fcurrent is CNA {
+    #'      loadRDS(fcurrent)
```

Comment:
you need to **assign** the result of a ```readRDS()``` call. This is different from a ```load()``` command.